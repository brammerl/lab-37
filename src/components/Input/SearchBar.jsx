import React from 'react';
import { setUserInfo, setUsername, setRepos} from '../../actions/reducerActions';
import { useDispatch, useUsername } from '../../hooks/githubContext';
import fetchUser from '../../services/fetchUser';
import { fetchRepos } from '../../services/fetchRepos';

const SearchBar = () => {
  const dispatch = useDispatch();
  const username = useUsername();
  
  const handleSubmit = (event) => {
    event.preventDefault();
    return Promise.all([fetchUser(username), fetchRepos(username)])
      .then(([userStats, repos]) => {
        dispatch(setUserInfo(userStats))
        dispatch(setRepos(repos))
      })
  }

  const handleChange = ({ target }) => {
    dispatch(setUsername(target.value))
  }
  
  return (
  <>
    <form onSubmit={handleSubmit}>
      <input type='text' value={username} onChange={handleChange}/>
      <button>Submit</button>
    </form>
  </>
  )
}

export default SearchBar;