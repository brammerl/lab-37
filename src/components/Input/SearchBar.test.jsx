import React from 'react';
import { shallow } from 'enzyme';
import SearchBar from './SearchBar';

jest.mock('../../hooks/githubContext')


describe('SearchBar', () => {
  let wrapper;
  let main; 

  beforeEach(() => {
    wrapper = shallow(<SearchBar />);
    main = wrapper.instance();
  })

  it('renders SearchBar', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
  