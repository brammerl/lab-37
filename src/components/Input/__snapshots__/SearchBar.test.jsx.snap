// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders SearchBar 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      type="text"
    />
    <button>
      Submit
    </button>
  </form>
</Fragment>
`;
